### PR TITLE
A0 2089 initialize forest properly

### PR DESCRIPTION
**Pull request by lesniak43**
# Description

https://cardinal-cryptography.atlassian.net/browse/A0-2089
Initializing the `Forest` properly.
Adding a hack to fight potential stall when too many non-finalized blocks are imported into the Substrate database. 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- I have added tests
- I have created new documentation

### CODE REVIEW

File: finality-aleph/src/sync/handler.rs [hunk 1]
```diff
@@ -105,19 +110,58 @@ impl<I: PeerId, J: Justification, CS: ChainStatus<J>, V: Verifier<J>, F: Finaliz
...
+                    if let ForestError::TooNew = e {
+                        warn!(
+                            target: LOG_TARGET,
+                            "There are more imported non-finalized blocks that can fit into the Forest: {}.", e
+                        );
+                        self.forest = forest;
+                        return Ok(());
+                    } else {
+                        return Err(Error::Forest(e));
+                    }
```
Comment (kostekIV) [inline finality-aleph/src/sync/handler.rs:158]: refactor to match

**Review by kostekIV — APPROVED** [review]
lgtm, small nits

File: finality-aleph/src/sync/handler.rs [hunk 2]
```diff
@@ -105,19 +110,59 @@ impl<I: PeerId, J: Justification, CS: ChainStatus<J>, V: Verifier<J>, F: Finaliz
...
+                        ForestError::TooNew => {
+                            warn!(
+                                target: LOG_TARGET,
+                                "There are more imported non-finalized blocks that can fit into the Forest: {}.", e
+                            );
+                            self.forest = forest;
+                            return Ok(());
+                        }
+                        _ => return Err(Error::Forest(e)),
+                    }
```
Comment (kostekIV) [inline finality-aleph/src/sync/handler.rs:159]: I was thinking of smth like this
```rust
match forest.update_body(header) {
   Err(ForestError::TooNew) => ...
   Err(e) => ...,
   _ => ...,
}
```
wdyt?

File: finality-aleph/src/sync/service.rs [hunk 3]
```diff
@@ -248,6 +249,9 @@ impl<
 
     /// Stay synchronized.
     pub async fn run(mut self) {
+        // TODO: Remove after finishing the sync rewrite.
+        let mut stall_ticker = interval(Duration::from_secs(30));
```
[the remark this answers is not in the document]
Comment (lesniak43) [reply finality-aleph/src/sync/service.rs]: fixed

File: finality-aleph/src/sync/handler.rs (hunk 2 above)
Comment (lesniak43) [reply finality-aleph/src/sync/handler.rs:159]: the last branch would be empty, right? I think it's better to leave `if let Err(e)`

Comment (lesniak43) [reply finality-aleph/src/sync/handler.rs]: so be it

**Pull request merged**